### ISSUE TITLE
Unable to upgrade/scale AAD integrated clusters

### ISSUE THREAD

**Issue by EPinci** [Azure/aks-engine#718]
**Is this a request for help?** Yes

---

**Is this an ISSUE or FEATURE REQUEST?**: This is an ISSUE.

---

**What version of acs-engine?**: v0.14.4

**Orchestrator and version**: Kubernetes v1.9.3

---

**What happened**:
Run acs-engine to upgrade a Kubernetes v1.9.3 cluster to v1.9.6.
Upgrade starts but errors out with:

```
[0m[0009] Error deleting agent VM k8s-nodepool1-93283987-0: No Auth Provider found for name "azure"
FATA[0009] Error upgrading cluster: No Auth Provider found for name "azure"
```

Rerunning the upgrade results in the same error leaving cluster partially upgraded and with an extra node.

**What you expected to happen**:
ACS-Engine to complete the cluster upgrade.

**How to reproduce it** (as minimally and precisely as possible):
Deploy a cluster with AAD integration and then run an upgrade.

**Anything else we need to know**:
I believe this applies also to scaling operations.

**Comment by CecileRobertMichon** [comment]
related to #717

**Comment by mboersma** [comment]
This was finally fixed by PR #1273. If you try AKS Engine v0.36.x or later, you shouldn't see this issue any longer.
